src/grt/grt-avhpi.adb

```diff
@@ -207,6 +286,10 @@ package body Grt.Avhpi is
             declare
                Sizes : Ghdl_Indexes_Ptr;
             begin
+               -- An explanation of what's going on here would be really
+               -- helpful.
+               -- It looks like we're finding out the size of the structure
+               -- but I've failed to understand how.
```

Comment:
Ah OK.  So that's the for_generate_statement.  There is an array containing the sub-instances, because the block can be repeated many times.  So we need to update the context to point to the corresponding sub-instance.